src/app/archive/snapp_tables.sql

```diff
@@ -93,17 +93,17 @@ CREATE TABLE snapp_updates
 );
 
 /* in OCaml, events are a list of array of field elements
-   here, a list is given by an id, each contained array is given its
+   here, a list is addressed by hash, each contained array is given its
     order within the list
 */
 CREATE TABLE snapp_events
-( list_id                  int              NOT NULL
+( list_hash                text             NOT NULL
```

Comment:
as discussed via PM, this table can be removed: the `events_list_hash` and `rollup_events_list_hash` can be replaced by an array of ints, representing a list of arrays, where each int is the id of a `snapp_state_data_array`

that eliminates a level of indirection, and is easier to understand, I'd offer